### PR TITLE
Create ContinuousIntegration.yml

### PR DESCRIPTION
**Pull request by jlconlin**

### CODE REVIEW

File: .github/workflows/ContinuousIntegration.yml
Comment (whaeck) [inline .github/workflows/ContinuousIntegration.yml]: So just an observation: you use make with a single processor and ctest with 2. Do we know what type of machine these are being run on? It would be nice to take advantage of multiprocessing if it is available - especially for the more time intensive projects.

Comment (jlconlin) [reply .github/workflows/ContinuousIntegration.yml]: Good point. I believe we have two processors at our disposal.

File: .github/workflows/ContinuousIntegration.yml
Comment (whaeck) [inline .github/workflows/ContinuousIntegration.yml:9]: You're doing this only for pull requests to master?

Elementary does this for pull requests to master but also push on branches:
```
# This workflow is triggered on pushes and pull requests to the repository.
on: 
  push:
    branches: '**'
  pull_request: 
    branches: 'master'
```

**Review by whaeck — CHANGES_REQUESTED** [review]

File: .github/workflows/ContinuousIntegration.yml
Comment (jlconlin) [reply .github/workflows/ContinuousIntegration.yml:9]: I copied what I had from the [RECONR](https://github.com/njoy/RECONR) repository. Thanks for a watchful eye.

**Review by whaeck — APPROVED** [review]
Good for me. Just wait until all actions have finished to merge it in.

**Pull request merged**
